Barotrauma/BarotraumaClient/ClientSource/GUI/GUITextBox.cs

```diff
@@ -617,10 +617,15 @@ public void ReceiveCommandInput(char command)
                 case '\b' when !Readonly: //backspace
                 {
                     using var _ = new TextPosPreservation(this);
-                    if (PlayerInput.KeyDown(Keys.LeftControl) || PlayerInput.KeyDown(Keys.RightControl))
+                    if (PlayerInput.IsCtrlDown())
                     {
-                        SetText(string.Empty, false);
-                        CaretIndex = Text.Length;
+                        var removeTo = FindStopSymbolLeft(Text, CaretIndex);
+                        var len = CaretIndex - removeTo;
+                        var newText = Text.Remove(removeTo, CaretIndex - removeTo);
+                        SetText(newText, false);
+                        CaretIndex = Math.Max(0, CaretIndex - len);
+                        CalculateCaretPos();
```

Comment:
Caret position seems to be incorrect after undoing and redoing.

Reply:
Thanks for review, @itchyOwl . 
Cursor position is always at the end of the input after undo/redo regardless of memento state before or after such action. I would agree that this is somewhat non-traditional behavior, but it will require non-trivial refactoring on how memento is working and focus of the PR was to add navigational shortcuts. I would prefer if this particual fix would be implemented in scope of different PR.

Here is a snippet of code for handling undo/redo that was not modified in PR:
```cs
case (char)0x1A when !Readonly && !SubEditorScreen.IsSubEditor(): // ctrl-z
    text = memento.Undo();
    if (text != Text)
    {
        ClearSelection();
        SetText(text, false);
        CaretIndex = Text.Length;
        OnTextChanged?.Invoke(this, Text);
    }
    break;
case (char)0x12 when !Readonly && !SubEditorScreen.IsSubEditor(): // ctrl-r
    text = memento.Redo();
    if (text != Text)
    {
        ClearSelection();
        SetText(text, false);
        CaretIndex = Text.Length;
        OnTextChanged?.Invoke(this, Text);
    }
    break;
```